Language/Classes/Constructors/Factories/return_type_A03_t01.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a dynamic type error if a factory returns a non-null object
```

Comment:
The updated version of this sentence (in [this PR](https://github.com/dart-lang/language/pull/2605)) does not include 'non-null'. The reason is that null never has a type which is a subtype of the actual return type (for all constructors except in the class `Null`, and we don't care exactly how that class is written because we couldn't write an equivalent one in regular Dart code anyway).

So we could have said "It is a dynamic type error if a factory returns null or a non-null object ...", and then we may just as well say "It is a dynamic type error if a factory returns an object ...".

I'm not quite sure how we can best handle this kind of situation (where the current spec differs from dart-lang/language#2605), but it is at least tempting to say that we should check the 2605 specification and see whether it is different from the corresponding part of the "current" specification, such that we don't create/update tests with obsolete (pre-nnbd) expectations. It could also be nice if we can avoid the need to update these `@assertion` paragraphs when 2605 is finally landed (and that's not going to be 2605 as-is, it's going to be several PRs where ).

So how could we proceed such that these things would work well for you?